### PR TITLE
add PopUp comp

### PR DESCRIPTION
**Pull request by Standupic**

### CODE REVIEW

File: src/Verktygsproffsen/Header.stories.fs [hunk 1]
```diff
@@ -1,2 +1,89 @@
...
+
+let iconTrack : string = importDefault "../svg/icon-truk.svg"
+
+type FormOrganization =
+    | Physical
+    | Legal
+
+type Face =
+    {
+      Face : FormOrganization
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: Please follow the same indentation everywhere: 4 spaces

File: src/Verktygsproffsen/Header.stories.fs [hunk 2]
```diff
@@ -1,2 +1,89 @@
...
+        let organizationFor face =
+            match face with
+            | Physical ->
+                "499"
+
+            | Legal ->
+                "625"
+
+        let getAdress face =
+            sprintf "FRI FRAKT ÖVER %s KR EXKL. MOMS" face
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: Good example and reminder that we need differently localized texts. We need to think how to better approach it. I guess we will need separate Json files for that.

File: src/Verktygsproffsen/Header.stories.fs [hunk 3]
```diff
@@ -1,2 +1,89 @@
...
+
+        let getStr param =
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: This function is unused. Remove it please

File: src/Verktygsproffsen/Header.stories.fs [hunk 4]
```diff
@@ -1,2 +1,89 @@
...
+                                          HTMLAttr.Custom ("htmlFor", "privat")
+                                          Id "privatLabel" ]
+                                    [ input [ Name "accessType"
+                                              Id "privat"
+                                              Type "radio"
+                                              Value "customer" ]
+                                      span [ Class "switch" ]
+                                        [ ]
+                                      span [ Class "name" ]
+                                        [ str "Privatperson" ] ]
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: This is a duplicate code of lines 72-83. Can be extracted to a separate function. Also, please don't copy the names and ids from the previous solution. E.g. `Id "private"` should have a better name. I would follow the same `b2c` and `b2b` approach, but with some suffix. The same applies to `privateLabel`.

File: src/sass/header/top.sass [hunk 5]
```diff
@@ -0,0 +1,103 @@
+.top
+	background: #473f3b
```
Comment (mvsmal) [inline src/sass/header/top.sass]: I would suggest to extract all colors to variables so we don't copy the same real color value to different places

File: src/sass/main.sass [hunk 6]
```diff
@@ -1,41 +1,9 @@
-@font-face
-    font-family: 'Kelp A3 Black Italic'
-    src: url('../../public/fonts/kelpta3-blackitalic-webfont.woff2') format('woff2'),
-    url('../../public/fonts/kelpta3-blackitalic-webfont.woff') format('woff'),
-    url('../../public/fonts/kelpta3-blackitalic-webfont.ttf') format('truetype'),
-    url('../../public/fonts/kelpta3-blackitalic-webfont.svg#kelpt_a3black_italic') format('svg')
-    font-weight: normal
-    font-style: normal
+@import './breakpoint/breakpoint'
```
Comment (mvsmal) [inline src/sass/main.sass]: Sass documentation recommends to use `@use` instead of `@import`. Let's switch to it already now.

File: src/Verktygsproffsen/Popup.stories.fs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+open Fable.Core.JsInterop
+
+let closeImg: string = importDefault "../svg/close.svg"
+
+type PopUpProps = { Text: string }
+
+let Popup =
+    FunctionComponent.Of(fun (props : PopUpProps) ->
+
+      let getText text =
```
Comment (mvsmal) [inline src/Verktygsproffsen/Popup.stories.fs]: I would either extract this function to the outer scope, or make it `inline`.

File: src/Verktygsproffsen/Popup.stories.fs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        div [ Class "popup-text" ] [str text]
+
+      div [ Class "popup" ]
+
+        [ div [ Class "popup-close" ]
```
Comment (mvsmal) [inline src/Verktygsproffsen/Popup.stories.fs]: We should think about the code style and follow it. With @atsapura we agreed that we should use
```fsharp
type SomeType =
    {
        SomeProp : string
    }
```
instead of
```fsharp
type SomeType =
    { SomeProp : string }
```
even though the latter looks more compact.
I think we should follow the same approach for lists (maybe only if a list contains multiple elements?). So like this:
```fsharp
div []
    [
        div [] []
        div [] []
    ]
```
But I am not sure here. @atsapura what do you think? @IvanSalov any comment from your side?

File: src/Verktygsproffsen/Header.stories.fs [hunk 9]
```diff
@@ -1,2 +1,89 @@
...
+open Fable.React
+open Fable.React.Props
+open PM.StyleGuide.Storybook
+open Fable.Core.JsInterop
+
+let iconTrack : string = importDefault "../svg/icon-truk.svg"
+
+type FormOrganization =
+    | Physical
+    | Legal
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: We have special term for this. Please change to 
```fsharp
type AccountType =
    | B2B
    | B2C
```

File: src/Verktygsproffsen/Header.stories.fs [hunk 10]
```diff
@@ -1,2 +1,89 @@
...
+
+type Face =
+    {
+      Face : FormOrganization
+    }
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: Let's follow the naming `SomeComponentProps`, in this case it should be:
```fsharp
type AccountTypeProps =
    {
        AccountType : AccountType
    }
````

File: src/sass/main.sass [hunk 11]
```diff
@@ -1,41 +1,9 @@
...
+@import './popup'
+@import './header/top'
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:400,600,700,800');
```
Comment (mvsmal) [inline src/sass/main.sass]: Shouldn't we better put it to `/.storybook/preview-head.html`? I believe with the current approach the font's will be downloaded and placed to the bundled application, but if we put it to head, we use a link to google fonts and this font might be already cached on the user's machine, so the load is faster.

File: src/Verktygsproffsen/Popup.stories.fs [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+            [ img [ Src closeImg ] ]
```
Comment (mvsmal) [inline src/Verktygsproffsen/Popup.stories.fs]: Can we make this icon as a background? What do you think about it?

File: src/Verktygsproffsen/Header.stories.fs (hunk 2 above)
Comment (Standupic) [reply src/Verktygsproffsen/Header.stories.fs]: Do you mean that all text value for different parts of the site we will be store into Json.file?

File: src/Verktygsproffsen/Header.stories.fs (hunk 3 above)
Comment (Standupic) [reply src/Verktygsproffsen/Header.stories.fs]: yeah sure.

File: src/Verktygsproffsen/Popup.stories.fs (hunk 8 above)
Comment (Standupic) [reply src/Verktygsproffsen/Popup.stories.fs]: I agree with you but when you have many props and children in the node of Html it very hard to follow described your structure.

Comment (atsapura) [reply src/Verktygsproffsen/Popup.stories.fs]: I don't have a strong opinion on this one. I guess if we don't update this block of code often (e.g. swapping lines) then we can afford compact style, since it is easy to read.

File: src/Verktygsproffsen/Header.stories.fs (hunk 4 above)
Comment (Standupic) [reply src/Verktygsproffsen/Header.stories.fs]: year of cause.

File: src/sass/main.sass (hunk 11 above)
Comment (martinholmin) [reply src/sass/main.sass]: +1

File: src/Verktygsproffsen/Header.stories.fs (hunk 1 above)
Comment (martinholmin) [reply src/Verktygsproffsen/Header.stories.fs]: @mvsmal let's get a .editorconfig file in place. Also can we add something like StyleCop to the CI process catch coding style deviations.

File: src/Verktygsproffsen/Header.stories.fs (hunk 2 above)
Comment (mvsmal) [reply src/Verktygsproffsen/Header.stories.fs]: Yes

File: src/Verktygsproffsen/Header.stories.fs [hunk 13]
```diff
@@ -1,2 +1,68 @@
...
+
+        let organizationFor account =
+            match account with
+            | B2C ->
+                "499"
+
+            | B2B ->
+                "625"
+
+        let getAdress account =
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: This is not address. This is "Free delivery for more than X kr". Let's rename to `freeDeliveryText`

File: src/Verktygsproffsen/Header.stories.fs [hunk 14]
```diff
@@ -1,2 +1,68 @@
...
+type AccountTypeProps =
+    { AccountType : AccountType }
+
+
+let Top =
+    FunctionComponent.Of(fun ( props : AccountTypeProps ) ->
+
+        let textValue =  [ "Privatperson"; "Företag"; ]
+
+        let organizationFor account =
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: This function should be named smth like `getMinTotalForFreeDelivery`. It is not organization. Also it should return `int` or `decimal`, definitely not a string. Then in the next function you should have `%d` or `%f` instead of `%s`. I would go for `int` and `%d`

File: src/Verktygsproffsen/Search.stories.fs [hunk 15]
```diff
@@ -0,0 +1,28 @@
+module PM.StyleGuide.Verktygsproffsen.Search
+
+open Fable.React
+open Fable.React.Props
+open PM.StyleGuide.Storybook
+
+// Should i use <option> type?
+// It will get rid of it will get rid of checking on an empty string in future?
+// also, allow us to manage behaviour when values no.
```
Comment (mvsmal) [inline src/Verktygsproffsen/Search.stories.fs]: Let's not do it now. The problem is that with `Option` type you can also have `Some ""` or even `Some null`. This is bad about .NET string type. But for now I would not overcomplicate it. Please remove the comments.

File: src/sass/header/search.sass [hunk 16]
```diff
@@ -0,0 +1,21 @@
+.header-search
+    width: 540px
+    position: relative
+    .form-input
+        border: none
+        width: 100%
+        border-radius: 5px
+        color: #3498db
```
Comment (mvsmal) [inline src/sass/header/search.sass]: Please use all colors as variables.

File: src/Verktygsproffsen/Search.stories.fs (hunk 15 above)
Comment (Standupic) [reply src/Verktygsproffsen/Search.stories.fs]: Ок

File: src/sass/header/search.sass (hunk 16 above)
Comment (Standupic) [reply src/sass/header/search.sass]: done
